src/ephy-bookmark.c

```diff
@@ -19,16 +19,31 @@
 
 #include "ephy-bookmark.h"
 
+#include "ephy-shell.h"
+#include "ephy-sync-crypto.h"
+#include "ephy-sync-utils.h"
+
+#include <string.h>
+
 struct _EphyBookmark {
   GObject      parent_instance;
 
   char        *url;
   char        *title;
   GSequence   *tags;
   gint64       time_added;
+
+  char        *id;
+  double       modified;
```

Comment:
Should probably leave a comment here (or somewhere) on why you needed double instead of float.